hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/hive/HoodieCombineHiveInputFormat.java

```diff
@@ -375,6 +389,45 @@ public InputSplit[] getSplits(JobConf job, int numSplits) throws IOException {
     // clear work from ThreadLocal after splits generated in case of thread is reused in pool.
     Utilities.clearWorkMapForConf(job);
 
+    // build internal schema for the query
+    if (job.getBoolean(INTERNAL_SCHEMA_CACHE_ENABLE, true)) {
+      if (result.size() > 0) {
+        Map<String, String> schemaMap = new HashMap<>();
+        ArrayList<String> uniqTablePaths = new ArrayList<>();
+        Arrays.stream(paths).forEach(path -> {
+          HoodieStorage storage = null;
+          try {
+            storage = new HoodieHadoopStorage(path.getFileSystem(job));
+            Option<StoragePath> tablePath = TablePathUtils.getTablePath(storage, HadoopFSUtils.convertToStoragePath(path));
```

Comment:
In which case there are multiple table paths?

Reply:
> In which case there are multiple table paths?

When we perform a query on two different Hudi tables, such as union, join, or subquery, there will be multiple table paths.